### ISSUE TITLE
NAV- Enlazar contrato Autoconsumo cuando hay baja de comercialziación

### ISSUE THREAD

**Issue by dgarciageo** [PKFEric/webProxima#909]
Cuando un contrato de comercialización que tiene autoconsumo hace un cambio que implica la baja de un contrato y el alta de otro, la ficha de autoconsumo "se pierde".

**Comment by JMunoz17** [comment]
Efectivamente, no esta implementado que al venir un M1_05 que implique dar de baja un contrato y crear uno nuevo se haga lo mismo con el contrato de autoconsumo. En cualquier caso yo creo que, al menos Belén, si que tenía constancia de esto (me suena haberlo comentado alguna vez).

Se puede hacer hacerlo, la cuestión es definir como.

Podemos directamente en el contrato de autoconsumo cambiar el código de contrato, fácil y rápido pero no se hasta que punto correcto porque pueden existir facturas asociadas a ese contrato de autoconsumo y la vez asociadas al contrato anterior.

La otra forma es dar de alta un nuevo contrato de autoconsumo donde necesitaríamos saber que valores respetar y cuales no y que tenemos que hacer con las fechas y con el detalle de financiación en caso de existir.

**Comment by bpesquero** [comment]
Creo que la mejor forma de hacerlo seriá la primera que propones, En ese caso, cuando existan facturas de autoconsumo asociadas al contrato anterior las seguiríamos teniendo asociadas en el detalle de financiación, no es así?
Antes de tomar una decisión me gustaría que analizaráis un poco las consecuencias de los diferentes procesos asociados, no vayamos a "liarla"

**Comment by jorgereal** [comment]
Hemos realizado la primera opción.

Al procesar un M1_02 con cambio de titular o un M1_05 con cambio de tarifa se cambia el cód. contrato en el contrato de autoconsumo al cód. contrato nuevo.

El cód. contrato en las líneas del detalle de autoconsumo se mantendrá, para llevar una trazabilidad correcta de qué documentos pertenecen a qué contrato. Pero todos se mostraran en la ficha del contrato de autoconsumo puesto que la relación se hace a través del campo "Cod. Detalle Financiación".

Tras revisar los procesos asociados hemos encontrado un inconveniente en el siguiente supuesto:

¿Podría darse el caso de que exista una factura sin registrar para un contrato dado de baja mediante un M1?
Al registrar una factura de un contrato de autoconsumo se busca su detalle para rellenar el "no. factura registrada", pero en este caso no se encontraría contrato de autoconsumo asociado y no se rellenaría esta información, dado que éste ya estaría asociado al nuevo contrato.

Para solucionar esto habría que modificar la búsqueda del contrato de autoconsumo. Partiendo de un contrato con el mismo cups que el de la factura pero con estado activado.

Confirmadnos si este caso podría darse para añadir esta modificación.

Subido a PRE.

**Comment by bpesquero** [comment]
Efectivamente el caso se puede dar. HAy que buscar por el contrato que corresponda para ese CUPS.

**Comment by jorgereal** [comment]
Se ha modificado el proceso de registro para contemplar el caso mencionado y buscar el contrato correspondiente por CUPS y estado activado.

Subido a PRE.

**Comment by bpesquero** [comment]
No sé muy bien cómo hacer las pruebas en PRE con los datos que tenemos, podemos subirlo a PRO con garantías?

gracias

**Comment by jorgereal** [comment]
Para probar este desarrollo habría que buscar un contrato de autoconsumo en PRE, generar un M1_01 con cambio de titular o cambio de tarifa y luego importar un M1_02 (en el caso de cambio de titular) o un M1_05 (en el caso de cambio de tarifa) generados a mano.

Nosotros ya hicimos una prueba con el contrato con el contrato COM/01822 que dio lugar al COM/03146, así que si queréis lo subimos.

**Comment by bpesquero** [comment]
Adelante

**Comment by jorgereal** [comment]
Subido a PRO.
